tests/dev_env_tests/test_get_datasets_per_data_model.py

```diff
@@ -0,0 +1,121 @@
+import json
+import subprocess
+import uuid
+
+import fasteners
+import pytest
+
+from mipengine.node_info_DTOs import NodeInfo
+from tests.dev_env_tests.nodes_communication import get_celery_app
+from tests.dev_env_tests.nodes_communication import get_celery_task_signature
+
+
+@pytest.fixture(autouse=True)
+def run_tests_sequentially():
+    with fasteners.InterProcessLock("semaphore.lock"):
+        yield
+
+
+def setup_data_table_in_db(node_id, datasets_per_data_model):
+    data_model_id = 2
+    dataset_id = 4
+    for data_model in datasets_per_data_model.keys():
+        print(data_model)
+        data_model_id += 1
+        dataset_id += 1
+        data_model_code, data_model_version = data_model.split(":")
+        sql_query = f"""INSERT INTO "mipdb_metadata"."data_models" VALUES ({data_model_id}, '{data_model_code}', '{data_model_version}', '{data_model}', 'ENABLED', null);"""
+        print(sql_query)
+        cmd = f'docker exec -i monetdb-{node_id} mclient db -s "{sql_query}"'
+        subprocess.call(cmd, shell=True)
+        for dataset_name in datasets_per_data_model[data_model]:
+            dataset_id += 1
+            sql_query = f"""INSERT INTO "mipdb_metadata"."datasets" VALUES ({dataset_id}, {data_model_id}, '{dataset_name}', '{dataset_name}', 'ENABLED', null);"""
+            print(sql_query)
+            cmd = f'docker exec -i monetdb-{node_id} mclient db -s "{sql_query}"'
+            subprocess.call(cmd, shell=True)
+
+
+# The cleanup task cannot be used because it requires specific table name convention
+# that doesn't fit with the initial data table names
+def teardown_data_tables_in_db(node_id):
```

Comment:
Instead of hardcoding the `dataset_id` to delete, it won't work when we change the datasets to 20, in the next PR, you can pass the data_models/datasets as a parameter and delete those specifically.